src/cocotb/__init__.py

```diff
@@ -432,3 +357,48 @@ def _stop_user_coverage() -> None:
         _user_coverage.stop()
         cocotb.log.debug("Writing coverage data")
         _user_coverage.save()
+
+
+def _setup_random_seed() -> None:
+    global _random_seed
+
+    seed_envvar = os.getenv("RANDOM_SEED")
+    if seed_envvar is None:
+        if "ntb_random_seed" in plusargs:
+            plusarg_seed = plusargs["ntb_random_seed"]
+            if not isinstance(plusarg_seed, str):
+                raise TypeError("ntb_random_seed plusarg is not a valid seed value.")
+            _random_seed = int(plusarg_seed)
```

Comment:
We don't have test coverage for these, and I don't know who uses these or what the values look like,
but using `int()` does change the behavior for some seed values.
If the seed is set to something like `0x345`, `eval()` would give an integer `837`, but `int()` raises a `ValueError`.

Reply:
There is `ast.literal_eval` and I could add another type check afterwards for `int`. `_random_seed` must be an `int` eventually.

Reply:
Sounds fine to me.